### PR TITLE
Add information about the NDEF records

### LINKED ISSUE #249: List overview of NDEF binary fields
Would probably make it a bit easier to talk about a lot of things in the spec

Something like (but drawn by us):

image

Also the below should probably have TNF as a separate column.

image

### PR DESCRIPTION
**Pull request by kenchris**
Fixes #249

### CODE REVIEW

**Comment by kenchris** [comment]
@Honry

File: index.html [hunk 1]
```diff
@@ -431,72 +457,238 @@
...
+      <ndef-record
+        header="0,1,0,1,0,TNF 6 (UNCHANGED)"
+        content="0,PAYLOAD LENGTH FOR THIS RECORD,_,_,_,*"
+        short>
+      </ndef-record>
+    </p>
+    <p>
+      Any implementation of Web NFC MUST transparently expose chunked records
+      as single logical records.
+    </p>
```
Comment (Honry) [inline index.html:707]: Missing a `</section>`.

Comment (kenchris) [reply index.html:707]: Will fix. Any other comment?

Comment (Honry) [reply index.html:707]: Reviewing, large contents. 😋

File: index.html [hunk 2]
```diff
@@ -431,72 +457,238 @@
...
+        short>
+      </ndef-record>
+    </p>
+  </section>
+  <section>
+    <h4>
+      External type records (TNF 4)
+    </h4>
+    <p>
+      External types are for client specified data and must have a type
```
Comment (Honry) [inline index.html]: Pls. add link for "External types".

File: index.html [hunk 3]
```diff
@@ -431,72 +457,238 @@
...
+      name following the NFC RTD standard.
```
Comment (Honry) [inline index.html]:
```suggestion
      name following the [[[NFC-RTD]]] standard.
```

File: index.html [hunk 4]
```diff
@@ -431,72 +457,238 @@
...
-    corresponding message structure that allows negotiation and activation of
-    an alternative communication carrier, such as Bluetooth or WiFi.
-    The negotiated communication carrier would then be used (separately) to
-    perform certain activities between the two devices, such as sending photos
-    to the other device, printing to a Bluetooth printer or streaming video to
-    a television set.
   </p>
+  <section>
+    <h4>
+      Empty NDEF records (TNF 0)
```
Comment (Honry) [inline index.html:480]: Can this be a `dfn`? Many `TNF=*` in Data mapping table could be linked to here.

File: index.html [hunk 5]
```diff
@@ -431,72 +457,238 @@
...
+  <section>
+    <h4>
+      External type records (TNF 4)
+    </h4>
+    <p>
+      External types are for client specified data and must have a type
+      name following the NFC RTD standard.
+    </p>
+    <p class=note>
+      Web NFC defined a special external type record that is not accessible
```
Comment (Honry) [inline index.html]: defines?

File: index.html [hunk 6]
```diff
@@ -378,20 +379,45 @@
...
+    chunk flag) whether the payload is chunked across multiple records.
+  </p>
+  <p class="note">
+    Web NFC turns all chunked records into logical records.
+  </p>
+  <p>
+    Bit 4 (SR, short record) indicates whether the record is a short record. A short
+    record is one with a payload length <= 255 bytes. Normal records can have payload
+    lengths exceeding 255 bytes up to a maximum of 4 GB. Short records only use one
+    byte to indicate length, whether as normal records use 4 bytes (2^32-1 bytes).
```
Comment (Honry) [inline index.html]: What does "(2^32-1 bytes)" mean here? I don't quite understand.

File: index.html (hunk 1 above)
Comment (kenchris) [reply index.html:707]: You probably want to render it :-)

File: index.html [hunk 7]
```diff
@@ -431,72 +457,238 @@
...
+       Well-known records (TNF 1)
+    </h4>
+    <p>
+      The NFC Forum has standardized a small set of useful data types for
+      use in NDEF records, for instance text, URL, and binary data such as
+      media. In addition, there are record types designed for more complex
+      interactions, such as smart poster, and handover records.
+    </p>
+    <p>
+      NFC Forum well-known type includes record types text, url, smart poster
```
Comment (Honry) [inline index.html]:
```suggestion
      NFC Forum <a>well-known type</a> includes record types text, url, smart poster
```

File: index.html [hunk 8]
```diff
@@ -378,20 +379,45 @@
...
+      content="*,PAYLOAD LENGTH - 1 (SR) to 4 bytes,ID LENGTH (optional),TYPE (optional),ID (optional),PAYLOAD (optional)">
+    </ndef-record>
+  </p>
+  <p>
+    Only the first three bytes (lines in figure) are mandatory. First the header,
+    followed by the type length and payload length, which may both be zero.
+  </p>
+  <p>
+    Bit 7 of the header (MB, message begin) indicates whether this is the beginning
+    of a message, the bit 6 (ME, message end) whether it is the end. Bit 5 (CF,
```
Comment (Honry) [inline index.html]:
```suggestion
    of a message, the bit 6 (ME, message end) indicates whether it is the end. Bit 5 (CF,
```

File: index.html [hunk 9]
```diff
@@ -378,20 +379,45 @@
...
+    chunk flag) whether the payload is chunked across multiple records.
```
Comment (Honry) [inline index.html]:
```suggestion
    chunk flag) indicates whether the payload is chunked across multiple records.
```

File: index.html [hunk 10]
```diff
@@ -378,20 +379,45 @@
...
+      header="*,*,*,*,*,*"
+      content="*,PAYLOAD LENGTH - 1 (SR) to 4 bytes,ID LENGTH (optional),TYPE (optional),ID (optional),PAYLOAD (optional)">
+    </ndef-record>
+  </p>
+  <p>
+    Only the first three bytes (lines in figure) are mandatory. First the header,
+    followed by the type length and payload length, which may both be zero.
+  </p>
+  <p>
+    Bit 7 of the header (MB, message begin) indicates whether this is the beginning
```
Comment (Honry) [inline index.html]: Why not just split definitions of each bits into separate paragraphs? BTW, these could be `dfn` as which had been referenced in somewhere of the spec.

File: index.html [hunk 11]
```diff
@@ -378,20 +379,45 @@
...
+    byte to indicate length, whether as normal records use 4 bytes (2^32-1 bytes).
+  </p>
+  <p>
+    Bit 3 (IL, Id length) indicates whether an ID length is available. This optional
+    identifier is a URL.
+  </p>
+  <p>
+    Bit 0-2 indicates the format of the type name, <dfn>TNF</dfn> (Type Name Format)
+    and is often exposed by native NFC software stacks. The field can take binary values
+    denoting the following NDEF record payload types:
```
Comment (Honry) [inline index.html]:
```suggestion
    denoting the following <a>NDEF record</a> payload types:
```

File: index.html [hunk 12]
```diff
@@ -431,72 +457,238 @@
...
-    The negotiated communication carrier would then be used (separately) to
-    perform certain activities between the two devices, such as sending photos
-    to the other device, printing to a Bluetooth printer or streaming video to
-    a television set.
   </p>
+  <section>
+    <h4>
+      Empty NDEF records (TNF 0)
+    </h4>
+    TNF 0 (empty) is used for empty records, which have no payload or type
```
Comment (Honry) [inline index.html]: Missing a full stop.

File: index.html [hunk 13]
```diff
@@ -431,72 +457,238 @@
...
+      <a>NDEF record</a> level payload identification, since <a>NDEF message</a>
+      level identification is used.
+    </p>
+  </section>
+  <section>
+    <h4>
+      Unknown type records (TNF 5)
+    </h4>
+    <p>
+      Unknown type records are records that store opaque data without
```
Comment (Honry) [inline index.html]:
```suggestion
      Unknown type records are records that store `opaque` data without
```

File: index.html [hunk 14]
```diff
@@ -431,72 +457,238 @@
...
+        </ndef-record>
+      </ndef-record>
+    </p>
+  </section>
+  <section>
+    <h4>
+        MIME type records (TNF 2)
+    </h4>
+    <p>
+      MIME type records are records that store opaque data with associated MIME type.
```
Comment (Honry) [inline index.html]:
```suggestion
      MIME type records are records that store `opaque` data with associated <a>MIME type</a>.
```

File: index.html [hunk 15]
```diff
@@ -431,72 +457,238 @@
...
+      <ndef-record
+        header="*,*,*,*,*,2 (MIME)"
+        content="*,*,*,SERIALIZED MIME TYPE,*,MIME TYPE PAYLOAD"
+        short>
+      </ndef-record>
+    </p>
+    <p class=note>
+      Web NFC has special handling for working with JSON MIME type data.
```
Comment (Honry) [inline index.html:556]:
```suggestion
      Web NFC has special handling for working with <a>JSON MIME type</a> data.
```

File: index.html [hunk 16]
```diff
@@ -431,72 +457,238 @@
...
+      level identification is used.
+    </p>
+  </section>
+  <section>
+    <h4>
+      Unknown type records (TNF 5)
+    </h4>
+    <p>
+      Unknown type records are records that store opaque data without
+      associated MIME type.
```
Comment (Honry) [inline index.html]:
```suggestion
      associated <a>MIME type</a>.
```

**Review by Honry — COMMENTED** [review]
Impressive! A great overview to know more about NDEF records.

**Comment by kenchris** [comment]
Fixed your comments @Honry Hopefully this won't cause too much rebasing issue for you

**Pull request merged**